packages/foam-core/src/note-graph.ts

```diff
@@ -1,121 +1,162 @@
-import { Graph, Edge } from 'graphlib';
-import { Position, Point } from 'unist';
-import GithubSlugger from 'github-slugger';
+import { Graph } from 'graphlib';
 import { EventEmitter } from 'events';
+import { Position, Point, URI, ID } from './types';
+import { uriToSlug, hashURI, getUriViaRelative } from './utils';
 
-type ID = string;
-
-export interface Link {
-  from: ID;
-  to: ID;
+export interface NoteSource {
+  uri: URI;
   text: string;
+  contentStart: Point;
+  end: Point;
+  eol: string;
 }
 
-export interface NoteLink {
-  to: ID;
+export interface WikiLink {
+  type: 'wikilink';
   text: string;
+  slug: string;
```

Comment:
yes that is a good point. in fact at the moment it is not, but the situation where this would be the case is:
`this is a [[the-greatest-day|great day]] for things`.
In the old code I was supporting this case too but I didn't port the feature to minimize scope creep. I forgot to fix the type, happy to do that if you guys think it might be better

Reply:
I removed `text` for now, we were not using and was causing confusion